src/foam/dao/FileJournal.js

```diff
@@ -281,27 +285,58 @@ foam.CLASS({
           .toString());
       `
     },
+    {
+      name: 'getEntry',
+      documentation: 'Reads a syntatically meaningful unit from the journal',
+      type: 'String',
+      args: [
+        {
+          name: 'reader',
+          type: 'BufferedReader'
+        }
+      ],
+      javaCode: `
+        try {
+          String line = reader.readLine();
+          if ( getMultiLine() ) {
+            StringBuilder sb = new StringBuilder();
+            while ( ! line.equals("})") ) {
+              sb.append(line);
+              line = reader.readLine();
+              if ( line == null )
```

Comment:
Surely this check needs to be earlier, we just called line.equals() but if line was null that would throw a NPE.